### PR TITLE
Rework the caching logic for subscription and entitlements 

### PR DESCRIPTION
**Pull request by miasma13**
Task/Issue URL: https://app.asana.com/0/414235014887631/1206909771441779/f

**Description**:

- Change default staleness time interval for subscription and entitlement cache to 20min
- When storing subscription in the cache use the date based on default interval or its date of renewalOrExpiry - whatever is earlier
- Make the VPN code use cache and its logic for doing fetch if data is stale
- On applicationDidBecomeActive refresh subscription details and entitlements
- In the app rely on subscription details and entitlements based on cache logic (cached if still valid else fetch)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/AppDelegate.swift [hunk 1]
```diff
@@ -435,7 +435,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
             }
 #endif
             SubscriptionPurchaseEnvironment.current = .appStore
-            await AccountManager().checkSubscriptionState()
+            await AccountManager().refreshSubscriptionAndEntitlements()
```
Comment (federicocappelli) [inline DuckDuckGo/AppDelegate.swift]: What about moving this from didFinishLaunchingWithOptions to applicationDidBecomeActive so we can refresh the cache every time the app comes back from the background? didFinishLaunchingWithOptions could not happen for days

**Review by federicocappelli — APPROVED** [review]
Just an idea in the comment, LGTM, ready for BSK update

File: DuckDuckGo/AppDelegate.swift (hunk 1 above)
Comment (quanganhdo) [reply DuckDuckGo/AppDelegate.swift]: +1 to having this invoked more often.

**Review by quanganhdo — APPROVED** [review]

File: DuckDuckGo/AppDelegate.swift (hunk 1 above)
Comment (miasma13) [reply DuckDuckGo/AppDelegate.swift]: There is already call below in `applicationWillEnterForeground` but will move it to `applicationDidBecomeActive`

**Pull request merged**